src/jackdaw/test/transports/rest_proxy.clj

```diff
@@ -0,0 +1,318 @@
+        (assoc client :subscription topics)))))
+
+(defn rest-proxy-poller
+  "Returns a function that takes a consumer and puts any messages retrieved
+   by polling it onto the supplied `messages` channel"
+  [messages]
+  (fn [consumer]
+    (let [{:keys [base-uri group-id instance-id]} consumer
+          url (format "%s/records" base-uri)
+          headers {"Accept" (content-types :byte-array)}
+          body nil]
+      (let [response @(handle-proxy-request http/get url headers body)]
+        (when (:error response)
+          (log/errorf "rest-proxy fetch error: %s" (:error response)))
+        (when (not (:error response))
+          (doseq [msg (:json-body response)]
+            (async/onto-chan messages [msg] false)))))))
+
+(defn rest-proxy-subscription
+  [config topic-metadata]
+  (-> (rest-proxy-client config)
+      (with-consumer)
+      (with-subscription topic-metadata)))
+
+(defn rest-proxy-consumer
+  "Creates an asynchronous Kafka Consumer of all topics defined in the
+   supplied `topic-metadata`
+
+   Puts all messages on the channel in the returned response. It is the
+   responsibility of the caller to arrange for the read the channel to
+   be read by some other process.
+
+   Must be closed with `close-consumer` when no longer required"
+  [config topic-metadata deserializers]
+  (let [continue?   (atom true)
+        xform       (comp
+                     #(apply-deserializers deserializers %)
+                     #(undatafy-record topic-metadata %))
+        messages    (async/chan 1 (map xform))
+        started?    (promise)
+        poll        (rest-proxy-poller messages)]
+
+    {:process (async/go-loop [consumer (rest-proxy-subscription config topic-metadata)]
+                (poll consumer)
+
+                (when-not (realized? started?)
+                  (deliver started? true)
+                  (log/infof "started rest-proxy consumer: %s" (proxy-client-info consumer)))
+
+                (if @continue?
+                  (do (poll consumer)
+                      (Thread/sleep 500)
+                      (recur consumer))
+                  (do
+                    (async/close! messages)
+                    (destroy-consumer consumer)
+                    (log/infof "stopped rest-proxy consumer: %s" (proxy-client-info consumer)))))
+     :started? started?
+     :messages messages
+     :continue? continue?}))
+
+(defn build-record
+  "Builds a Kafka Producer and assoc it onto the message map"
+  [m]
+  ;; The easiest way to make this as real as possible is to build
+  ;; a real ProducerRecord and then get extract the data from it. This way
+  ;; we don't have to calculate the partition oureslves.
+  ;;
+  ;; On the other hand, this means we have a bit of an unwanted dependency
+  ;; (in this ns) on the Kafka Client API. Perhaps later it might be
+  ;; worth calculating this partition ourselves based on the supplied
+  ;; topic-metadata but there are a few wrinkles to iron out with that
```

Comment:
The wrinkles mentioned here relate to the cleanup of the write command suggested [above]( https://github.com/FundingCircle/jackdaw/pull/28/files#r246106743). We want a data representation of a write that both transports can interpret.